catalyst_voices/apps/voices/lib/widgets/modals/voices_share_dialog.dart

```diff
@@ -0,0 +1,217 @@
+import 'package:catalyst_voices/common/ext/ext.dart';
+import 'package:catalyst_voices/widgets/modals/details/voices_align_title_header.dart';
+import 'package:catalyst_voices/widgets/snackbar/voices_snackbar.dart';
+import 'package:catalyst_voices/widgets/snackbar/voices_snackbar_type.dart';
+import 'package:catalyst_voices/widgets/widgets.dart';
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:catalyst_voices_localization/catalyst_voices_localization.dart';
+import 'package:catalyst_voices_shared/catalyst_voices_shared.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+enum ShareType {
+  clipboard('Clipboard'),
+  xTwitter('X / Twitter'),
+  linkedin('LinkedIn'),
+  facebook('Facebook'),
+  reddit('Reddit');
+
+  final String name;
+
+  const ShareType(this.name);
+
+  SvgGenImage get icon => switch (this) {
+        ShareType.clipboard => VoicesAssets.icons.duplicate,
+        ShareType.xTwitter => VoicesAssets.icons.xTwitter,
+        ShareType.linkedin => VoicesAssets.icons.linkedin,
+        ShareType.facebook => VoicesAssets.icons.facebook,
+        ShareType.reddit => VoicesAssets.icons.reddit,
+      };
+
+  String description(VoicesLocalizations l10n, String itemType) =>
+      switch (this) {
+        ShareType.clipboard => l10n.shareDirectLinkToItem(itemType),
+        ShareType.xTwitter => l10n.shareLinkOnSocialMedia(itemType, name),
+        ShareType.linkedin => l10n.shareLinkOnSocialMedia(itemType, name),
+        ShareType.facebook => l10n.shareLinkOnSocialMedia(itemType, name),
+        ShareType.reddit => l10n.shareLinkOnSocialMedia(itemType, name),
+      };
+
+  String shareMessage(VoicesLocalizations l10n) => switch (this) {
+        ShareType.clipboard => l10n.shareOnSocialMedia(name),
+        ShareType.xTwitter => l10n.shareOnSocialMedia(name),
+        ShareType.linkedin => l10n.shareOnSocialMedia(name),
+        ShareType.facebook => l10n.shareOnSocialMedia(name),
+        ShareType.reddit => l10n.shareOnSocialMedia(name),
+      };
```

Comment:
Can't it be simplified to this?

```suggestion
  String shareMessage(VoicesLocalizations l10n) => l10n.shareOnSocialMedia(name);
```